ouroboros-consensus-cardano-test/test/Test/ThreadNet/Cardano.hs

```diff
@@ -488,8 +490,39 @@ mkProtocolCardanoAndHardForkTxs
               generatedSecretsByron
               propPV
 
-    pInfo :: ProtocolInfo m (CardanoBlock c)
+    pInfo :: ProtocolInfo (CardanoBlock c)
     pInfo = protocolInfoCardano
+              paramsByron
```

Comment:
Now that I've seen these parameter duplication happen in so many places, I'm wondering if introducing a

```
foo :: ... -> (ProtocolInfo blk, m [BlockForging m blk]
```

scheme and defining/using that throughout would be a worthwhile investment to cut down on repetition at the various use sites.

The `protocolInfo*` functions used to serve this "all of the config" purpose. But this PR splits that into a slimmer `protocolInfo*` and the new `blockForging*` alongside it. So the proposal here is to re-introduce a `protocolInfoAndBlockForging*` so that we again have an "all of the config" combinator, which ought to save some noise at the numerous sites where we do in fact need both.

Reply:
This pattern would also reduce the duplication I'm seeing in the new definitions of `protoclInfo*` and `blockForging*` functions: you're having to project out the same config parameters, etc, in each definition. Instead, you could essentially rename the existing `protocolInfo*` to `protocolInfoAndBlockForging*`, and munge its return type and body top-level expression from `ProtocolInfo{x,y,z} to `(y, ProtocolInfo{x,z})` and then define the new `protoclInfo*` and `blockForging*` as easy projections of `protocolInfoAndBlockForging*`.

Reply:
Another possible benefit: would the various calls to `plainTestNodeInitialization` be easier to update (just the renaming `protocolInfo*` to `protocolInfoAndBlockForging*`) if we changed its single argument type to match the return type of `protocolInfoAndBlockForging*`?

Reply:
That would've been simpler, but I thought that the next step should be to pass the configuration which allows to construct `BlockForging` not `BlockForging` itself - `BlockForging`.  Isn't `BlockForging` in `cardano-node` an abstraction leak?

Reply:
When we pass configuration then it's possible to block the listener until it actually receives something different than the current configuration (although that's right now is not that easy, because many types are missing `Eq` instances).

Reply:
Even if the pass in values affecting the `BlockForging` that change instead of passing in the `BlockForging` itself, we'll still need to have a function somewhere that constructs the `BlockForging` value, right? So I think the de-duplication opportunity would still apply there, wouldn't it?

Reply:
Fixed: now the functions return a tuple `(ProtocolInfo blk, m [BlockForging m blk])`; the slightly annoying thing is that chaining a function `f` which produces such a tuple with `fst` leaves `GHC` puzzled how to instantiate `m`.

Reply:
>chaining a function `f` which produces such a tuple with `fst` leaves GHC puzzled how to instantiate `m`

Ah, yes, that is a bother